src/pymorize/scripts/update_dimensionless_mappings.py

```diff
@@ -0,0 +1,169 @@
+or special unit conversions, and updates the dimensionless_mappings.yaml file with
+proper formatting including standard names as comments.
+
+The script keeps any existing values defined in the YAML file and leaves new values empty,
+preserving human-defined special cases.
+"""
+
+import glob
+import json
+import os
+import re
+
+import click
+import yaml
+
+from ..core.logging import logger
+
+# Units patterns that indicate dimensionless or special conversion variables
+DIMENSIONLESS_PATTERNS = [
+    r"1$",  # Pure dimensionless
+    r"\d+\s?%",  # Percentage
+    r"0.001",  # Salinity scaling factor
+]
+
+# Special keywords to check in units
+SPECIAL_KEYWORDS = [
+    "mol",  # May need carbon or other element specifications
+    "%",  # Percentage units
+]
+
+
+def is_dimensionless_unit(unit):
+    """Check if a unit string represents a dimensionless quantity"""
+    for pattern in DIMENSIONLESS_PATTERNS:
+        if re.match(pattern, unit):
+            return True
+    return False
+
+
+def extract_variables_from_tables(tables_path):
+    """
+    Extract all variables with dimensionless units or special units from CMIP6 tables.
+    Returns a dictionary of variable_name: {unit: unit, standard_name: standard_name}
+    """
+    variables = {}
+
+    # Get all JSON files in the tables directory
+    table_files = glob.glob(os.path.join(tables_path, "CMIP6_*.json"))
+
+    for table_file in table_files:
+        try:
+            with open(table_file, "r") as f:
+                table_data = json.load(f)
+
+            # Skip files without variable entries
+            if "variable_entry" not in table_data:
+                continue
+
+            for var_name, var_info in table_data["variable_entry"].items():
+                # Skip if no units defined
+                if "units" not in var_info:
+                    continue
+
+                unit = var_info["units"]
+
+                # Check if this is a dimensionless unit or contains special keywords
+                is_special = is_dimensionless_unit(unit) or any(
+                    keyword in unit for keyword in SPECIAL_KEYWORDS
+                )
+
+                if is_special:
+                    standard_name = var_info.get("standard_name", "not_specified")
+
+                    # Only add if not already in our dictionary or if this has a standard_name and previous doesn't
+                    if var_name not in variables or (
+                        standard_name != "not_specified"
+                        and variables[var_name]["standard_name"] == "not_specified"
+                    ):
+
+                        variables[var_name] = {
+                            "unit": unit,
+                            "standard_name": standard_name,
+                        }
+        except Exception as e:
+            logger.error(f"Error processing {table_file}: {e}")
+
+    return variables
+
+
+def update_yaml_file(yaml_path, variables):
```

Comment:
Why did you create the yaml dump implementation by hand? We have ruamel.yaml in the package, you can just add the comments for standand name (which I love!) as an EOL comment. I forgot the exact syntax, but have a look in esm-tools, it's in there somewhere.

Reply:
Answer from @mandresm in the chat: "It's from Windsurf". So, `git blame` to any portion that sees this should proudly declare "The robots are at fault!"

Reply:
Honestly because I did not remember how to add comments in ruamel objects. It was not robot's fault, I thought about it too, but I was lazy... I am happy to improve that and use ruamel.